doc: add workaround for the USB installation issue (#432)
harvester/docs: docs/install/usb-install.md

There are a couple of ways to create a USB installation flash drive.

:::caution
**Known Issue:** For the v1.2.0 ISO image, there is a known [issue](https://github.com/harvester/harvester/issues/4510) where the interactive ISO installation will get stuck using the USB method.

> Refer to the [workaround documentation](#harvester-interactive-iso-hangs-with-the-usb-installation-method) for detailed information.

Reviewer: ```suggestion
Refer to the [Harvester interactive ISO hangs with the USB installation method](#harvester-interactive-iso-hangs-with-the-usb-installation-method) for details and a workaround.
```

### balenaEtcher

:::caution

There is a known issue refer to [[BUG] v1.2.0 Interactive ISO Fails to Install On Some Bare-Metal Devices](https://github.com/harvester/harvester/issues/4510).

You can check the details [here](#harvester-installation-hang-on-some-bare-metal-devices)

> :::

Author: good suggestion, updated!

- Plug the USB flash drive into a USB 2.0 slot.
- For version `v0.3.0` or above, remove the `console=ttyS0` parameter when booting. Press `e` to edit the GRUB menu entry and remove the `console=ttyS0` parameter.

### Harvester interactive ISO hangs with the USB installation method

> During installation from a USB flash drive with v1.2.0 ISO image (created by tools like `balenaEtcher`, `dd`, etc.), the installation process may get stuck on the initial image loading process. This happens because a required label is missing on the boot partition. Therefore, the installation cannot mount the data partition correctly, causing some checks in "dracut" to be blocked.
> 
> If you encounter this issue, you'll observe a screen similar to the screenshot below, and the process will hang for at least 50 minutes (the default timeout value from `dracut`).

Reviewer: ```suggestion
During installation from a USB flash drive with v1.2.0 ISO image (created by tools like `balenaEtcher`, `dd`, etc.), the installation process may get stuck on the initial image loading process because a required label is missing on the boot partition. Therefore, the installation cannot mount the data partition correctly, causing some checks in `dracut` to be blocked.

If you encounter this issue, you'll observe the following similar output, and the process will hang for at least 50 minutes (the default timeout value from `dracut`).
```

![](/img/v1.2/install/usb-install-hang.png)

> To address this problem, you can manually modify the root partition as follows:

Reviewer: ```suggestion
#### Workaround

To address this problem, you can manually modify the root partition as follows:
```

```text

> # Replace the `CDLABEL=COS_LIVE` with your USB data partition. (Usually, it is the first partition with the device  name `sdx` that hangs on your screen)
> $linux ($root)/boot/kernel cdroot root=live:CDLABEL=COS_LIVE rd.live.dir=/ rd.live.squashimg=rootfs.squashfs console=tty1 console=ttyS0 rd.cos.disable net.ifnames=1 <-- original
> # $linux ($root)/boot/kernel cdroot root=live:/dev/sda1 rd.live.dir=/ rd.live.squashimg=rootfs.squashfs console=tty1 console=ttyS0 rd.cos.disable net.ifnames=1 <-- modified

Reviewer: ```suggestion
# Replace the `CDLABEL=COS_LIVE` with your USB data partition. Usually, your USB data partition is the first partition with the device name `sdx` that hangs on your screen.
# Original
$linux ($root)/boot/kernel cdroot root=live:CDLABEL=COS_LIVE rd.live.dir=/ rd.live.squashimg=rootfs.squashfs console=tty1 console=ttyS0 rd.cos.disable net.ifnames=1
# Modified 
$linux ($root)/boot/kernel cdroot root=live:/dev/sda1 rd.live.dir=/ rd.live.squashimg=rootfs.squashfs console=tty1 console=ttyS0 rd.cos.disable net.ifnames=1
```

``` 

> The modified parameter should look like the figure below:
> ![](/img/v1.2/install/grub-parameter-modified.png)

Reviewer: ```suggestion
The modified parameter should look like the following:

![](/img/v1.2/install/grub-parameter-modified.png)
```

- Plug the USB flash drive into a USB 2.0 slot.
- For version `v0.3.0` or above, remove the `console=ttyS0` parameter when booting. Press `e` to edit the GRUB menu entry and remove the `console=ttyS0` parameter.

> ### Harvester installation hang on some Bare-Metal devices

Reviewer 2: IIUC, this bug is not related to bare-metal devices but caused by invalid partition label and all USB installation method will be impacted?

> ### Harvester installation hang on some Bare-Metal devices

Author: agree, update the title!

If a USB flash v1.2.0 ISO through `balenaEtcher`, `dd`, or something similar ISO editor. The installation will hang on some bare-metal devices.
That is because the label of `boot` partition was lost. If the installation can not mount the data partition correctly, some checking of `dracut` will hang.

If you are getting this issue, you will see something like the following screenshot, and it will hang for at least 50 minutes. (This timeout value comes from `dracut`)
![](/img/v1.2/install/usb-install-hang.png)

To workaround this issue, you could manually modify the root partition as below.
```text

> # replace the `CDLABEL=COS_LIVE` with your USB data partition. (Usually, it is the device of the above situation, and the first partition is the data partition)

Reviewer 2: ```suggestion
# Replace the `CDLABEL=COS_LIVE` with your USB data partition. (Usually, it is the first partition with the device  name `sdx` that hangs on your screen)
```

$linux ($root)/boot/kernel cdroot root=live:CDLABEL=COS_LIVE rd.live.dir=/ rd.live.squashimg=rootfs.squashfs console=tty1 console=ttyS0 rd.cos.disable net.ifnames=1 <-- original
# $linux ($root)/boot/kernel cdroot root=live:/dev/sda1 rd.live.dir=/ rd.live.squashimg=rootfs.squashfs console=tty1 console=ttyS0 rd.cos.disable net.ifnames=1 <-- modified
``` 

> The result looks like the following figure.

Reviewer 2: ```suggestion
The modified parameter should look like the figure below:
```

![](/img/v1.2/install/grub-parameter-modified.png)

> After that, press `Ctrl + x` to boot. You should enter the installer as usual.

Reviewer 2: ```suggestion
After making this adjustment, press `Ctrl + x` to initiate booting. You should now enter the installer as usual.
```
